controllers/loadtest_controller.go

```diff
@@ -287,6 +287,9 @@ func (r *LoadTestReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		adjustAvailabilityForDefaults(status.DefaultServerPool, defaultServerPool)
 
 		for pool, requiredNodeCount := range missingPods.NodeCountByPool {
+			if requiredNodeCount == 0 {
```

Comment:
I think this should be changed in the logic of `adjustAvailabilityForDefaults` instead. If `c == 0`, then we don't make changes to the node count by pool. Otherwise, if `defaultPoolName` is empty, we return an error; if it is not empty, then we add the count to the default pool count.

Reply:
Done